### DISCUSSION TITLE
Can the icons for LAN IP addresses be customized?

### DISCUSSION THREAD

**Discussion by rayangl** [dani-garcia/vaultwarden#5080] [Q&A]
![image](https://github.com/user-attachments/assets/83676f64-9d0a-4f1a-a5fc-ce92e98b56b9)
![image](https://github.com/user-attachments/assets/46fb4376-4009-4661-90d7-28b660e8d0d7)

**Comment by BlackDex** [comment]
Yes, you can. If you crate a file file called `10.10.10.1.png`, or what ever the domain is in the first URL you configured that is possible.
Though, if a `.miss` file exists, that takes precedent over the `.png` file if the lifetime of the icon is expired. That might be something we should adjust to fallback to an existing file even though there is a `.miss` maybe.

But besides that, if you create such a file it will use that, but the cache timeout will be looked at. You can either refresh the time of those icon files with `touch` via a script maybe. Or increase the timeouts here:
https://github.com/dani-garcia/vaultwarden/blob/cd195ff24339056bc5246bd5ee9106379e7fb6e6/.env.template#L320-L325

**Reply by alx-xlx** [reply]
I tried but it doesn't work. Maybe it is disabled ?
None of the icons for `<IP>.png` works
Even though after deleting the .miss and touch all of them, restart service etc 

<img width="1775" height="763" alt="OzyWUJwFaz" src="https://github.com/user-attachments/assets/27a6881c-03fa-45b1-89ef-dd8270608493" />

**Reply by stefan0xC** [reply]
`HTTP_REQUEST_BLOCK_NON_GLOBAL_IPS` is enabled by default so you want to set this setting to `false` https://github.com/dani-garcia/vaultwarden/blob/5d84f17600e179280e44c391b92ee9eecc2b7cdc/.env.template#L353-L355

**Reply by alx-xlx** [reply]
yes that worked :)

<img width="146" height="55" alt="YHzbn3vkKi" src="https://github.com/user-attachments/assets/0857aa38-7a15-4873-9d8c-60e9590cc2bc" />

**Comment by J4ker** [comment]
hey what about the PORT ?

I have different services on different port, can we assign different images based on ports ?

IPADDRESS.png

**Reply by BlackDex** [reply]
Nope. The clients do not provide schemas, ports or paths during these requests. 

The only way you could do it is by either using a reverse proxy and provide unique FQDN's per item. Or, create custom icon cache files, set the first URL in the item to match that name, set it to never match for that URL, and set the second URL as matching item. Since only the first URL item is used to search for an icon.

**Comment by rayangl** [comment]
I configured the icon cache location and files, but the client still couldn't read them. The cache files were properly generated in the specified directory with correct permissions, yet the client interface kept showing placeholder icons instead of loading the cached ones. This behavior persisted even after restarting both the server and client applications.
<img width="1920" height="1040" alt="image" src="https://github.com/user-attachments/assets/5a00b57a-3798-44e7-8bd3-af6691f0482e" />
<img width="1181" height="356" alt="image" src="https://github.com/user-attachments/assets/9ff38a72-93ad-4ae1-864c-24c370d6c408" />
<img width="979" height="689" alt="image" src="https://github.com/user-attachments/assets/1edf4985-87c0-46ad-affe-25a18c4513ca" />

**Reply by BlackDex** [reply]
You need to remove the `ICON_CACHE_FOLDER` env, since that points to a location which does not exists within the container is my guess. If you do not define it, it will use the default `/data/icon_cache`
